contracts.md

```diff
@@ -1,42 +1,42 @@
 
 <a name="contracts-vs-facades"></a>
-### Contracts Vs. Facades
+### مقارنة واجهات كونتراكتس (Contracts) مع واجهات فَسادس (Facades)
 
-Laravel's [facades](/docs/{{version}}/facades) and helper functions provide a simple way of utilizing Laravel's services without needing to type-hint and resolve contracts out of the service container. In most cases, each facade has an equivalent contract.
+توفر واجهات لارافيل [فَسادس (Facades)](/docs/{{version}}/facades) والتوابع المساعدة (helper functions) أسلوباً بسيطاً للاستفادة من خدمات لارافيل بدون الحاجة للتلميح إلى النوع (type-hint) أو استبيان واجهات كونتراكتس (contracts) من حاوي الخدمات (service container). وفي معظم الحالات، يكون لكل واجهة فَساد (Facade) واجهة كونتراكت (Contract) مقابلة مساوية لها. 
 
-Unlike facades, which do not require you to require them in your class' constructor, contracts allow you to define explicit dependencies for your classes. Some developers prefer to explicitly define their dependencies in this way and therefore prefer to use contracts, while other developers enjoy the convenience of facades. **In general, most applications can use facades without issue during development.**
+على عكس واجهات فَساد (Facade)، التي لا تحتاج أن منك تضمنها في الباني (constructor) الخاص بصفك (class)، واجهات كونتراكتس (contracts) تسمح لك بتعريف الاعتماديات (dependencies) بشكل صريح لصفك (class). فبعض المطورون يفضلون تعريف الاعتماديات (dependencies) بشكل صريح بهذه الطريقة فبالتالي يستخدمون واجهات كونتراكتس (contracts)، بينما يفضل البعض الآخر السهولة والراحة التي تقدمها واجهات فَساد (Facade). **بشكل عام، يمكن لمعظم التطبيقات استخدام واجهات فَسادس (Facades) بدون مشاكل خلال عملية التطوير.** 
 
 <a name="when-to-use-contracts"></a>
-## When To Use Contracts
+## متى تُستخدم واجهات كونتراكنتس
 
-The decision to use contracts or facades will come down to personal taste and the tastes of your development team. Both contracts and facades can be used to create robust, well-tested Laravel applications. Contracts and facades are not mutually exclusive. Some parts of your applications may use facades while others depend on contracts. As long as you are keeping your class' responsibilities focused, you will notice very few practical differences between using contracts and facades.
+يعود قرار استخدام واجهات كونتراكتس (contracts) أو واجهات فَسادس (Facades) لذوقك الشخصي الخاص أو للذوق الشخصي لفريق التطوير. فيمكن استخدامها كل منهما لبناء تطبيقات لارافيل قوية ومُختبرة بشكل جيد. وإن استخدام أي منهما لا يستبعد إمكانية استخدام الآخر. فيمكن لبعض أجزاء تطبيقك استخدام واجهات فَسادس (Facades) بينما تعتمد الأجزاء الأخرى على واجهات كونتراكتس (contracts). وطالما أنك تبقي مسؤوليات الصفوف مركزة، فإنك لن تلاحظ سوى اختلافات تطبيقية قليلة بين استخدام واجهات فَسادس (Facades) و استخدام واجهات كونتراكتس (contracts). 
```

Comment:
I don't think that "يعود لذوقم الشخصي" is the best translation, because the development process It is not made according to tastes "اذواق", but rather according to use "استخدام", definitely here what is intended to be used

Reply:
Noted